src/components/notification/inline-notification.ts

```diff
@@ -83,11 +83,56 @@ const iconsForKinds = {
  */
 @customElement(`${prefix}-inline-notification`)
 class BXInlineNotification extends FocusMixin(LitElement) {
+  /**
+   * Current timeout identifier
+   */
+  protected _timeoutID: number | null = null;
+
   /**
    * Notification type.
    */
   protected _type = NOTIFICATION_TYPE.INLINE;
 
+  /**
+   * Cancels the current timeout configured for the notification
+   * @param timeoutID current timeout's identifier
+   */
+  protected _cancelTimeout(timeoutID: number) {
+    clearTimeout(timeoutID);
+    this._timeoutID = null;
+  }
+
+  /**
+   * Overrides (if exists) the timeout to close the notification
+   * @param timeout the time in ms
+   */
+  protected _initializeTimeout(timeout: number) {
+    if (this._timeoutID) {
+      this._cancelTimeout(this._timeoutID);
+    }
+    this._timeoutID = setTimeout(this._handleTimerInitiatedClose.bind(this), timeout);
+  }
+
+  /**
+   * Handles timeout close request of this modal.
+   */
+  protected _handleTimerInitiatedClose(triggeredBy: undefined) {
```

Comment:
Wondering if you wanted to make `undefined` as the default value, instead of making it the type of `triggeredBy`? If so, we can:

```suggestion
  protected _handleTimerInitiatedClose(triggeredBy?: Element) {
```

Reply:
This was to clarify how should I handle the callbacks. With the last refactorization should be clearer.